include/tsar/Analysis/Memory/MemorySet.h

```diff
@@ -439,6 +385,37 @@ class MemorySet {
     return IsChanged;
   }
 
+  /// \brief Clarify information about the memory locations in this set.
+  ///
+  /// \param [in] From List of pairs of memory locations, on which
+  /// clarification of information will occur. The first location of the pair is 
+  /// the original memory location, the second location is the aggregated memory 
+  /// location. If the second location does not have an exact intersection with 
+  /// any location of this set, the first location will be added to this set.
+  template<class Ty> void clarify(
+      llvm::SmallVectorImpl<std::pair<Ty, Ty>> &From) {

```

Comment:
```suggestion
      llvm::ArrayRef<std::pair<Ty, Ty>> &From) {
```

Т.к. не нужно добавлять новые участки памяти в коллекцию `From`, то `ArrayRef` лучше подходит, т.к. позволяет в качестве параметра указвать любой массиво-подобный тип данных.

Reply:
Что-то у меня не получается после замены на `ArrayRef` вызвать `clarify()`. При компиляции выдается сообщение о том, что `SmallVector` не наследуется от `ArrayRef`.

Reply:
Забыл убрать в прототипе `&` (т.к. `ArrayRef` - сам по себе ссылка на массив) и должен быть все-таки изменяемый тип `MutableArrayRef`:

```suggestion
      llvm::MutableArrayRef<std::pair<Ty, Ty>> From) {
```
Еще видимо проблемы вывода типа будут, т.к. `clarify` - шаблонный, придется вызов явно `.clarify<MemoryLocationRange>(...)` писать.

Тогда пожалуй можно прототип не менять, чтобы вызов не усложнялся.